src/ttsim/interface_dag.py

```diff
@@ -107,12 +109,12 @@ def main(
         if isinstance(n, InterfaceFunction) and qn not in input_qnames
     }
 
-    # If main_targets are None, all failures and warnings are included, anyhow.
-    if fail_and_warn and main_targets is not None:
-        main_targets = include_fail_and_warn_nodes(
-            functions=functions,
-            explicit_main_targets=main_targets,  # type: ignore[arg-type]
-        )
+    main_targets = include_fail_or_warn_nodes(
```

Comment:
Just a note that I moved the branching logic into the function. No need to think about that when looking at the body of main.